asciidoc-confluence-publisher-client/src/main/java/org/sahli/asciidoc/confluence/publisher/client/http/ConfluencePage.java

```diff
@@ -21,22 +21,28 @@
  */
 public class ConfluencePage {
 
+    private final String ancestorId;
```

Comment:
Included ancestorId because I needed it for the replace strategy.

Reply:
I again analysed the need for reading back the `ancestorId`  and passing in when updating the page: 

- For all non-root pages, the ancestor id is defined by the content id of the parent page, so there is no need to use the ancestor id read from the existing page.

- When using the `APPEND_TO_ANCESTOR` publish strategy, the ancestor id of the root page is defined by the configuration.

- The only case the ancestor id from the existing page is currently used, is when using the `REPLACE_ANCESTOR`. But in this case, the hierarchy of the ancestor page should not be changed, so the ancestor id parameter does not even have to be passed to the confluence api, so `null` could be passed in this case. Unfortunately, currently, the `HttpRequestFactory.updatePageRequest()` method expects the `ancestorId` parameter to be present, which is no longer true in all situations.

Therefore, I would opt for making the `ancestorId` parameter  in `HttpRequestFactory.updatePageRequest()` optional, not passing it to the `PagePayload` via the builder. That way, the `ancestorId` does not need to be read back from existing pages (and no magic handling for dealing with potential multiple ancestor is needed).